src/components/Message/MessageIsThreadReplyInChannelButtonIndicator.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useRef } from 'react';
+import type { LocalMessage } from 'stream-chat';
+import { formatMessage } from 'stream-chat';
+import {
+  useChannelActionContext,
+  useChannelStateContext,
+  useMessageContext,
+  useTranslationContext,
+} from '../../context';
+
+export const MessageIsThreadReplyInChannelButtonIndicator = () => {
+  const { t } = useTranslationContext();
+  const { channel } = useChannelStateContext();
+  const { openThread } = useChannelActionContext();
+  const { message } = useMessageContext();
+  const parentMessageRef = useRef<LocalMessage | null | undefined>(undefined);
+
+  const querySearchParent = () =>
+    channel
+      .getClient()
+      .search({ cid: channel.cid }, { id: message.parent_id })
+      .then(({ results }) => {
+        if (!results.length) return;
+        parentMessageRef.current = formatMessage(results[0].message);
+      });
+
+  useEffect(() => {
+    if (
+      parentMessageRef.current ||
+      parentMessageRef.current === null ||
+      !message.parent_id
+    )
+      return;
+    const localMessage = channel.state.findMessage(message.parent_id);
+    if (localMessage) {
+      parentMessageRef.current = localMessage;
+      return;
+    }
+  }, [channel, message]);
+
+  if (!message.parent_id) return null;
+
+  return (
+    <div className='str-chat__message-is-thread-reply-button-wrapper'>
+      <button
+        className='str-chat__message-is-thread-reply-button'
+        data-testid='message-is-thread-reply-button'
+        onClick={async () => {
+          if (!parentMessageRef.current) {
+            // search query is performed here in order to prevent multiple search queries in useEffect
+            // due to the message list 3x remounting its items
+            await querySearchParent();
+            if (parentMessageRef.current) {
+              openThread(parentMessageRef.current);
+            } else {
+              // prevent further search queries if the message is not found in the DB
+              parentMessageRef.current = null;
+            }
+            return;
+          }
+          openThread(parentMessageRef.current);
+        }}
+      >
+        {t<string>('Thread reply')}
+      </button>
```

Comment:
To prevent surprises with forms, can you please add a `type="button` prop to the `<button />` element?

Also, please add error handling around `await querySearchParent()`, as in case it fails, the failure will be reported as "Unhandled Promise Rejection" in customer apps, and there is nothing they can do about it.

Reply:
Also, didn't we get rid of the generics of `t<string>()`?